src/cunumeric/index/advanced_indexing_omp.cc

```diff
@@ -78,8 +50,15 @@ struct AdvancedIndexingImplBody<VariantKind::OMP, CODE, DIM, OUT_TYPE> {
         sizes[tid] += static_cast<int64_t>(index[p] && ((idx + 1) % skip_size == 0));
       }
     }  // end of parallel
-    size_t size = thrust::reduce(thrust::omp::par, sizes.begin(), sizes.end(), 0);
-    thrust::exclusive_scan(thrust::omp::par, sizes.begin(), sizes.end(), offsets.begin());
+    size_t size = 0;
+    for (auto idx = 0; idx < max_threads; ++idx) {
+      size += sizes[idx];
+      offsets[idx] = sizes[idx];
```

Comment:
You're already computing a reduction over `sizes`, so you might as well just do the exclusive prefix sum by setting `offsets[idx] = size` and moving to before you increment size.

Reply:
that is right. Thank you!